src/compiler/types.ts

```diff
@@ -1807,8 +1807,8 @@ namespace ts {
 
     export interface CatchClause extends Node {
         kind: SyntaxKind.CatchClause;
-        parent?: TryStatement;
-        variableDeclaration: VariableDeclaration;
+        parent?: TryStatement; // We make this optional to parse missing try statements
+        variableDeclaration?: VariableDeclaration;

```

Comment:
Can you update `createCatchClause` and `updateCatchClause` in _factory.ts_ and add `| undefined` to the `variableDeclaration` parameters in both cases? This is needed for those using the Compiler API with `-strictNullChecks` enabled for their project.